src/platformClient.js

```diff
@@ -8,11 +9,13 @@ class PlatformClient {
    * Constructor
    * @param {Function} logFunction log function, defaults to console.log
    * @param {Function} tokenResolverFunction optional token resolver function, if provided it will extend the request headers with Bearer token
+   * @param {Object} axiosDefaults optional defaults object that will be merged with axios defaults
    */
-  constructor(logFunction, tokenResolverFunction = null) {
+  constructor(logFunction, tokenResolverFunction = null, axiosDefaults = {}) {
```

Comment:
It would be better to just take in a client instead, that way we don't need any special logic to handle things like **I want to add an interceptor**

Reply:
Good point, added.